### ISSUE TITLE
Change parent split ratio when left child is focused

### ISSUE THREAD

**Issue by matthewdias** [koekeishiya/kwm#301]
Often I find myself wanting to change the split ratio of a node, but can't, because it has child nodes, and their split ratio takes the controls. This could be solved by changing it so that when the left node is focused, it changes the split ratio for the parent's pair, and when the right node is focused, it changes for the child pair. Eg:
```
          a          
         / \         	If 2 is focused, 1/b will change.
        1   b        	If 3 is focused, 2/3 will change.
           / \       
          2   3      
                     
                     

---------------------
| SubTree |         |
|    1    |    2    |
|         |         |
|         |---------|
|         |         |
|         |    3    |
|         |         |
---------------------
```

**Comment by koekeishiya** [comment]
> Often I find myself wanting to change the split ratio of a node, but can't, because it has child nodes, and their split ratio takes the controls

Agreed, just have not been able to decide the best method for doing this. Your solution would not work as a general purpose solution with the following tree, which is what I want to come up with before doing any changes in this part of the code.

```
              a
          /        \
        b          c
      /    \       /    \
   d       3   e      6
  /   \         /   \
1     2     4    5
```

1 - Left child of 'd' changes size of 'b'
2 - Right child of 'd' changes size of 'd' 
3 - Right child of 'b' changes size of 'b'
4 - Left child of 'e' changes size of 'c'
5 - Right child of 'e' changes size of 'e'
6 - Right child of 'c' changes size of 'c'

No way to change size of 'a', and with more complex bsp-trees this will have a greater impact.
Now you can argue that more complex trees are unlikely to happen, but this depends a lot on the displays a user has access to and I'd like to avoid any assumptions.

**Comment by dominiklohmann** [comment]
Is there a way to enable users to change the split in a direction? You would have to programmatically figure out what North/East/South/West mean tho. I feel like this would be the easiest to use.

**Comment by koekeishiya** [comment]
> Is there a way to enable users to change the split in a direction? You would have to programmatically figure out what North/East/South/West mean tho.

Pretty sure this would be ambiguous. How do you translate a binary-tree into North/East/South/West.

**Comment by matthewdias** [comment]
Hm, it might be possible to do it with directions if you determined the window in the direction the same way you do for focusing, and then walk the tree to find the parent they have in common. There's probably a scenario im missing where that wouldnt work tho. It also would probably be too slow if you're expecting it to happen on a keypress. 

Doing it by depth is probably a better solution, if there was some kind of way to visualize it. I was planning to eventually make a sort of tree visualizer for kwm that could pop up on demand, or when a change in the tree is made, so that might be helpful. 

In lieu of that, maybe there could be a split prefix, that launches you into a split editing mode. There you could select the parents of nodes, and change split values, and there could be an accompanying border to tell you which parent you have selected.

Just spewing out thoughts here... Maybe I'm getting too crazy. They're just windows xD

**Comment by koekeishiya** [comment]
> Hm, it might be possible to do it with directions if you determined the window in the direction the same way you do for focusing, and then walk the tree to find the parent they have in common. There's probably a scenario im missing where that wouldnt work tho. It also would probably be too slow if you're expecting it to happen on a keypress. 

Traversing a tree of this size should not be noticable in my opinion, and the proposed solution here would work all trees. It is still not intuitive without knowing how the bsp-tree is constructed. I'll create and put up a screenshot in a sec.

**Comment by matthewdias** [comment]
Yeah, I'd be fine with it, but it might be hard to grok. The "resize mode" I proposed earlier would probably be better for ease of use, but would take more steps, and would be kind of cumbersome.

>In lieu of that, maybe there could be a split prefix, that launches you into a split editing mode. There you could select the parents of nodes, and change split values, and there could be an accompanying border to tell you which parent you have selected.

In such a mode, you'd probably have to have two new commands to select parent, and select first child for resizing.

**Comment by koekeishiya** [comment]
I'm gonna implement directional behaviour when I have time. I really do not feel like having to create any form of graphical interface for this.

**Comment by koekeishiya** [comment]
This has now been implemented to allow directions. The old command is still available for backwards compatibility. The new commands is the same as the old one, with the addition of `north | south | west | east` as an additional argument. E.g: `kwmc window -c expand 0.05 east`

The following is a screenshot representing the bsp-tree from my earlier post, and descriptions of what the result for the above directions would yield in this scenario.

<img width="1680" alt="bsp" src="https://cloud.githubusercontent.com/assets/6175959/14441128/9e91fdc0-0033-11e6-89e0-7ddf1f50266b.png">

Screenshot of a window-layout represented by the bsp-tree I listed above. I forgot to draw the "master root node", which I refer to as 'a'. Applying directions to find a given window and finding the LCA:

window 1 has focus:
north -> invalid
south -> finds window 3 -> adjust split-ratio of 'b'
east -> finds window 2 -> adjust split-ratio of 'd'
west -> finds window 5 -> adjust split-ratio of 'a'

window 2 has focus:
north -> invalid
south -> finds window 3 -> adjust split-ratio of 'b'
east -> finds window 4 -> adjust split-ratio of 'a'
west -> finds window 1 -> adjust split-ratio of 'd'

window 6 has focus:
north -> finds window 4 -> adjust split-ratio of 'c'
south -> invalid
east -> invalid
west -> finds window 3 -> adjust split-ratio of 'a'
